lib/rouge/lexers/powershell.rb

```diff
@@ -652,28 +97,47 @@ class Powershell < Shell
         rule %r/\s+/, Text
         rule %r/\$?"/, Str::Double, :double_quotes
         rule %r/\$'/, Str::Single, :ansi_string
-
         rule %r/'/, Str::Single, :single_quotes
-
         rule %r/\*/, Keyword
-
         rule %r/;/, Text
         rule %r/[^=\*\s{}()$"'`<]+/, Text
         rule %r/\d+(?= |\Z)/, Num
         rule %r/</, Text
         mixin :interp
       end
 
+      state :hashtable do 
+        rule %r/\w+/, Name::Variable
+        rule %r/}/, Operator, :pop!
+        mixin :root
+      end
+
+      state :multiline do
+        rule %r/\.(#{MULTILINEKEYWORDS})/i, Keyword::Pseudo
+        rule %r/[\w\s\.\-\,:\/{}<>"*]/, Comment
+        rule %r/#>/, Comment, :pop!
+        mixin :root
+      end
+
+      state :heredoc do
+        rule %r/"@/, Operator, :pop!
+        rule %r/[^$\n]+/, Str::Heredoc
+        rule %r/[$]/, Str::Heredoc
+        mixin :data
+      end
+
       prepend :basic do
-        rule %r(<#[\s,\S]*?#>)m, Comment::Multiline
-        rule %r/#.*$/, Comment::Single
+        rule %r(#requires\s-version \d.\d*$),Comment::Preproc
+        rule %r(<#), Comment, :multiline
+        rule %r(@{), Operator, :hashtable
+        rule %r(@"), Operator, :heredoc
         rule %r/\b(#{OPERATORS})\s*\b/i, Operator
-        rule %r/\b(#{ATTRIBUTES})\s*\b/i, Name::Attribute
-        rule %r/\b(#{KEYWORDS})\s*\b/i, Keyword
+        rule %r/\b(#{ATTRIBUTES})\s*\b/i, Name::Builtin::Pseudo
+        rule %r/[a-z,A-Z,0-9]+?-[a-z,A-Z,0-9]*/, Generic::Strong
+        rule %r/\b(#{KEYWORDS})\b/i, Keyword
         rule %r/\b(#{KEYWORDS_TYPE})\s*\b/i, Keyword::Type
         rule %r/\bcase\b/, Keyword, :case
-        rule %r/\b(#{BUILTINS})\s*\b(?!\.)/i, Name::Builtin
       end
     end
   end
-end
+end
```

Comment:
Re-introduce deleted newline.

```suggestion

end
```

Reply:
Is this some kind of style preference?

Reply:
Yes. But if anything else, this is a superfluous change that can be avoided.

Reply:
For future reference, I am okay with this @ashmaroli. When people clean up a bit on their way to helping us out, I usually welcome it.